plugin/src/software/aws/toolkits/eclipse/amazonq/lsp/AmazonQLspClientImpl.java

```diff
@@ -575,4 +578,72 @@ private boolean isUriInWorkspace(final String uri) {
             return false;
         }
     }
+
+    @Override
+    public final void sendPinnedContext(final Object params) {
+        Activator.getLogger().info("Pinned Context: sendPinnedContext called with params: " + params);
+        
+        // Get current active editor and add textDocument information
+        IEditorPart activeEditor = null;
+        try {
+            var workbench = PlatformUI.getWorkbench();
+            if (workbench != null) {
+                var activeWindow = workbench.getActiveWorkbenchWindow();
+                if (activeWindow != null) {
+                    var activePage = activeWindow.getActivePage();
+                    if (activePage != null) {
+                        activeEditor = activePage.getActiveEditor();
+                    }
+                }
+            }
+        } catch (Exception e) {
+            Activator.getLogger().warn("Could not get active editor: " + e.getMessage());
+        }
+        
+        Object updatedParams = params;
+        if (activeEditor != null && activeEditor instanceof ITextEditor) {
```

Comment:
this logic is also the same as previous comment. Same comment applies here.If needed try to use or extend existing logic to achieve the same instead of de-dup in